lighthouse-core/audits/seo/tap-targets.js

```diff
@@ -0,0 +1,319 @@
+ * @returns {LH.Audit.TapTargetTableItem[]}
+ */
+function getTableItems(overlapFailures) {
+  const tableItems = overlapFailures.map(
+    ({
+      tapTarget,
+      overlappingTarget,
+      tapTargetScore,
+      overlappingTargetScore,
+      overlapScoreRatio,
+    }) => {
+      const largestCR = getLargestRect(tapTarget.clientRects);
+      const width = Math.floor(largestCR.width);
+      const height = Math.floor(largestCR.height);
+      const size = width + 'x' + height;
+      return {
+        tapTarget: targetToTableNode(tapTarget),
+        overlappingTarget: targetToTableNode(overlappingTarget),
+        size,
+        width,
+        height,
+        tapTargetScore,
+        overlappingTargetScore,
+        overlapScoreRatio,
+      };
+    });
+
+  tableItems.sort((a, b) => {
+    return b.overlapScoreRatio - a.overlapScoreRatio;
+  });
+
+  return tableItems;
+}
+
+/**
+ * @param {LH.Artifacts.TapTarget} target
+ * @returns {LH.Audit.DetailsRendererNodeDetailsJSON}
+ */
+function targetToTableNode(target) {
+  return {
+    type: 'node',
+    snippet: target.snippet,
+    path: target.path,
+    selector: target.selector,
+  };
+}
+
+class TapTargets extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'tap-targets',
+      title: 'Tap targets are sized appropriately',
+      failureTitle: 'Tap targets are not sized appropriately',
+      description:
+        'Interactive elements like buttons and links should be large enough (48x48px), and have enough space around them, to be easy enough to tap without overlapping onto other elements. [Learn more](https://developers.google.com/web/fundamentals/accessibility/accessible-styles#multi-device_responsive_design).',
+      requiredArtifacts: ['Viewport', 'TapTargets'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    const hasViewportSet = ViewportAudit.audit(artifacts).rawValue;
+    if (!hasViewportSet) {
+      return {
+        rawValue: false,
+        explanation:
+          'Tap targets are too small because of a missing viewport config',
+      };
+    }
+
+    const tooSmallTargets = getTooSmallTargets(artifacts.TapTargets);
+    const overlapFailures = getAllOverlapFailures(tooSmallTargets, artifacts.TapTargets);
+    const uniqueOverlapFailures = mergeSymmetricFailures(overlapFailures);
+    const tableItems = getTableItems(uniqueOverlapFailures);
+
+    const headings = [
+      {key: 'tapTarget', itemType: 'node', text: 'Tap Target'},
+      {key: 'size', itemType: 'text', text: 'Size'},
+      {key: 'overlappingTarget', itemType: 'node', text: 'Overlapping Target'},
+    ];
+
+    const details = Audit.makeTableDetails(headings, tableItems);
+
+    const tapTargetCount = artifacts.TapTargets.length;
+    const failingTapTargetCount = new Set(overlapFailures.map(f => f.tapTarget)).size;
```

Comment:
maybe add a quick comment that we intentionally want to use `overlapFailures` and not `uniqueOverlapFailures`? I could see myself in 2 years looking at this code and thinking it was a bug without the context :)

Reply:
Good point! I renamed `uniqueOverlapFailures` to `overlapFailuresForDisplay`, do you think that also works?